ambry-utils/src/main/java/com/github/ambry/utils/Throttler.java

```diff
@@ -85,6 +85,14 @@ public void maybeThrottle(double observed) throws InterruptedException {
     }
   }
 
+  /**
+   * Update desiredRatePerSec for this throttler.
+   * @param desiredRatePerSec the new desiredRatePerSec
+   */
+  public void updateDesiredRatePerSecond(double desiredRatePerSec){
+    this.desiredRatePerSec = desiredRatePerSec;
```

Comment:
feel like here needs a lock, as you can see `desiredRatePerSec` is used in three places in `maybeThrottle()` method. Without being protected by `lock`, it could be changed meanwhile `maybeThrottle()` is being executed.